src/Build/Logging/FancyLogger/ANSIBuilder.cs

```diff
@@ -14,6 +15,35 @@ public static string ANSIRemove(string text)
             return Regex.Replace(text, "\\x1b(?:[@-Z\\-_]|\\[[0-?]*[ -\\/]*[@-~])", "");
         }
 
+        public static int ANSIBreakpoint(string text, int position)
+        {
+            if (position >= text.Length) return text.Length;
+            // Get substring
+            string substring = text.Substring(0, position);
```

Comment:
Can we just iterate along the string until we either reach position or find the right number of non-ANSI characters? That should let you get by without allocating any new strings.